graph/src/components/ethereum/adapter.rs

```diff
@@ -1,8 +1,18 @@
 use ethabi::{Bytes, Error as ABIError, Event, Function, LogParam, ParamType, Token};
 use failure::{Error, SyncFailure};
 use futures::{Future, Stream};
+use std::collections::HashSet;
+use std::iter::FromIterator;
 use web3::error::Error as Web3Error;
-use web3::types::{Address, Block, BlockId, BlockNumber, H256};
+use web3::types::*;
+
+use super::types::*;
+
+/// A collection of attributes that (kind of) uniquely identify an Ethereum blockchain.
+pub struct EthereumNetworkIdentifiers {
```

Comment:
The plural to me suggests that these are identifiers for multiple networks. How do you feel about using singular here?

Reply:
Ah shoot. Fair point. Yeah, I was trying to find a way to convey the fact that it can't be trusted as a unique identifier. I'm happy with a rename if you think that (or something else) is clearer

Reply:
I do 😉